### PR TITLE
TIAF Python Test Seeding: Moved/Previously Reviewed Changes.

### PR DESCRIPTION
**Pull request by jonawals**
Signed-off-by: John <jonawals@amazon.com>

## What does this PR do?

This PR adds the source changes that have been moved and thus previously reviewed needed for the Python test seeding to a staging branch.

### CODE REVIEW

File: Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Include/Static/TestImpactRuntimeConfigurationFactory.h [hunk 1]
```diff
@@ -8,8 +8,23 @@
 
 #include <TestImpactFramework/TestImpactConfiguration.h>
 
+#include <AzCore/JSON/document.h>
+
 namespace TestImpact
 {
+
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Include/Static/TestImpactRuntimeConfigurationFactory.h:15]: This has been moved from `TestImpactRuntimeConfigurationFactory.cpp`.

File: Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Source/TestImpactCommandLineOptions.cpp [hunk 2]
```diff
@@ -32,10 +32,12 @@ namespace TestImpact
             TestFailurePolicyKey,
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Common/Code/Source/TestImpactCommandLineOptions.cpp:32]: These changes have been moved from `TestImpactNativeCommandLineOptions.cpp`.

File: Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestRunner/Common/TestImpactTestEnumerator.h [hunk 3]
```diff
@@ -45,61 +45,75 @@ namespace TestImpact
             AZStd::optional<AZStd::chrono::milliseconds> enumerationTimeout,
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestRunner/Common/TestImpactTestEnumerator.h:45]: These changes have been moved from a free func to a member func.

File: Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestRunner/Common/TestImpactTestRunnerBase.h [hunk 4]
```diff
@@ -33,47 +33,66 @@ namespace TestImpact
         //! @param clientCallback The optional client callback to be called whenever a run job changes state.
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/TestRunner/Common/TestImpactTestRunnerBase.h:33]: These changes have been moved from a free func to a member func.

File: Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/Target/Native/TestImpactNativeTargetListCompiler.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/Target/Native/TestImpactNativeTargetListCompiler.cpp:1]: These changes have been moved from `TestImpactTargetListCompiler.cpp`.

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Include/TestImpactFramework/Python/TestImpactPythonRuntime.h [hunk 6]
```diff
@@ -8,37 +8,143 @@
...
-        {
-
-        }
-
-        //! Returns true if the runtime has test impact analysis data (either preexisting or generated).
-        bool HasImpactAnalysisData() const
-        {
-            return false;
-        }
+            const AZStd::vector<ExcludedTarget>& testsToExclude,
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Runtime/Python/Code/Include/TestImpactFramework/Python/TestImpactPythonRuntime.h:33]: These changes have been copied from `TestImpactNativeRuntime.h`.

**Review by moraaar — APPROVED** [review]

**Review by hultonha — APPROVED** [review]

**Pull request merged**